src/connect/render.cpp

```diff
@@ -126,8 +126,11 @@ namespace {
                 // These are not included in the fingerprint as they are changing a lot.
                 if (printing) {
                     JSON_FIELD_INT("time_printing", params.print_duration) JSON_COMMA;
-                    JSON_FIELD_INT("time_remaining", params.time_to_end) JSON_COMMA;
                     JSON_FIELD_INT("progress", params.progress_percent) JSON_COMMA;
+                    //Send only valid time_to_end value
+                    if(params.time_to_end != (uint32_t)-1){
+                        JSON_FIELD_INT("time_remaining", params.time_to_end) JSON_COMMA;
+                    }
```

Comment:
Style nitpicks:
* The whole JSON_* macro monster is excluded from auto-formatting because indentation. Nevertheless, the rest of that we probably might want to follow, and that includes an empty space between `if` and the parens.
* This is C++, so we shall either prefer C++-style casts (`static_cast<uint32_t>(-1)`) or, even better, `numeric_limits<uint32_t>::max()`.

Reply:
I am just looking into similar time estimate problems on XL (-1 interpreted as regular time) and there is even `TIME_TO_END_INVALID` constant defined in
https://github.com/prusa3d/Prusa-Firmware-Buddy-Private/blob/private/src/common/marlin_server_shared.h#L9
using the numeric limits.

Reply:
Thanks for the note. I have not noticed the constant as it was not used in any code relevant to my task and therefore I had no idea someone have already introduced it. I was thinking of doing the same thing.

I'll also use it in other places that have been using (uint32_t) casting until now.